components/automate-ui/src/app/pages/project/rules/project-rules.component.ts

```diff
@@ -84,21 +84,20 @@ export class ProjectRulesComponent implements OnInit, OnDestroy {
             : false;
         })).subscribe();
 
-      this.store.select(routeParams).pipe(
-        pluck('ruleid'),
-        filter(identity),
-        takeUntil(this.isDestroyed))
-        .subscribe((id: string) => {
-          this.store.dispatch(new GetRule({ id }));
-        });
-
-      this.store.select(routeParams).pipe(
-        pluck('id'),
-        filter(identity),
-        takeUntil(this.isDestroyed))
-        .subscribe((id: string) => {
-          this.store.dispatch(new GetProject({ id }));
-        });
+      combineLatest(
+        this.store.select(routeParams).pipe(pluck('id'), filter(identity)),
+        this.store.select(routeParams).pipe(pluck('ruleid'), filter(identity))
+      ).pipe(
+        takeUntil(this.isDestroyed),
+        map(([pId, rId]) => {
```

Comment:
You can include types with destructured objects. So if you use this...
```suggestion
        map(([project_id, rule_id]: [string, string]) => {
```
you can then delete L93-94.